### PR TITLE
validation and docs after refactor

### PR DESCRIPTION
**Pull request by Fersoil**

### CODE REVIEW

File: R/classes-plate.R [hunk 1]
```diff
@@ -53,40 +60,117 @@ is_valid_data_type <- function(data_type) {
 #'
 #' @importFrom R6 R6Class
 #'
+#' @export
```
[the remark this answers is not in the document]
Comment (Fersoil) [reply R/classes-plate.R:63]: I guess not exporting the function would result in user not having access to the documentation, thats why I added the @export statement

[the remark this answers is not in the document]
Comment (nizwant) [reply R/classes-plate.R:63]: yeah that makes sense

File: R/classes-plate_builder.R [hunk 2]
```diff
@@ -22,7 +27,7 @@ PlateBuilder <- R6::R6Class(
     #' @description
     #' Initialize the PlateBuilder object
     #'
-    #' @param plate_name - plate name obtained from filename
+    #' @param batch_name - plate name obtained from the plate info
```
[the remark this answers is not in the document]
Comment (Fersoil) [reply R/classes-plate_builder.R:30]: I do agree that the `batch_name` is an optional field. I am not sure if the `plate_name` should be extracted by the plate builder though. I will make this field simply optional in the plate builder.

File: R/parser.R [hunk 3]
```diff
@@ -1,5 +1,5 @@
 unified_datatypes <- c("Median", "Count", "Net MFI", "Avg Net MFI", "Mean")
```
[the remark this answers is not in the document]
Comment (Fersoil) [reply R/parser.R:1]: good point

File: R/classes-plate.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (Fersoil) [reply R/classes-plate.R:63]: but actually, when the `@export` field is ommited, the library behaviour is as follows:
1. you can actually access the docs
2. you can create the plate using the plate builder
3. you can't create the Plate object for yourself

I'll leave it without the `@export` field then

**Review by nizwant — APPROVED** [review]

**Pull request merged**